scripts.js

```diff
@@ -0,0 +1,266 @@
+"use strict";
+
+let state = {curPage: 1, numOfJoke: 10, curJokeLink : null, firstNumOfPageNav: 1}; 
+const ICNDB_URl = "http://api.icndb.com/jokes/";
+const DADDY_URL = "https://icanhazdadjoke.com/";
+
+let curPage = 1; 
+let numOfJoke = 10;
+let curJokeLink = null;
+let firstNumOfPageNav = 1;
+let isAtHome = true;
+$(".pagination").hide();
+$("#loading").hide();
+$("#icndb").click(renderHomePage);
+$("#daddy").click(renderHomePage);
+$("#assport").click(renderHomePage);
+$("#next-btn").click(changeToNextPageNav);
+$("#prev-btn").click(changeToPrevPageNav);
+$(".page-btn").click(loadNextPage);
+$("#project-description").click(displayProjectInfo);
+$(".paper").hide();
+$("#home").click(resetToHome);
+
+ /**
+   * hide div elements on the default page.
+   * display loading image alarming users that
+   * their joke is loading.
+*/
+function hideHomeItems() {
+    $(".title").hide();
+    $(".intro").hide();
+    $("#loading").show();
+}
+
+/**
+ * call is made when user is currently on About Project page
+ * and clicks home button.home-display-joke.
+ * hides all elements to transform view to default home page
+ */
+function displayProjectInfo() {
+    hideHomeItems();
+    $(".user-option-container").hide();
+    $(".home-display-joke").empty();
+    $(".pagination").hide();
+    $("footer").hide();
+    setTimeout(() => {
+        $("#loading").hide();
+        $(".paper").show();
+        $("footer").show();
+    }, 1000);
+}
+
+/**
+ * Call is made when user clicks Home btn 
+ * on the top navigation
+ * reload the page
+ * users will the default page
+ */
+function resetToHome() {
+    location.reload();
+}
+
+/**
+ *  function is called when user click one 
+ *  of the joke option
+ *  hides items items displayed on the home page
+ *  identifies the joke the user clicked
+ * @param {event} returns the event triggered by
+ * the option user chose on the form
+ */
+function renderHomePage(event) {
+    hideHomeItems();
+    $(".pagination").hide()
+    $(".chosen-joke").remove();
+    $(".home-display-joke").empty();
+    $("footer").hide();
+    $("home-display-joke").hide();
+    if (event.target.id === "icndb") {
+        curJokeLink = "icndb";
+        icndbFetch();
+    } else if (event.target.id === "daddy") {
+        curJokeLink = "daddy";
+        daddyFetch();
+    } 
+}
+
+/**
+ * this function identifies the type of joke
+ * that user selected and make appropriate fetch call
+ */
+function checkJokeTypeForFetch() {
```

Comment:
Functions that starts with the word `check` is usually a bad naming. You description here is misleading as well. What this function actually do is to fetch a set of jokes from the designated page.